cicecore/cicedyn/infrastructure/io/io_pio2/ice_pio.F90

```diff
@@ -529,6 +530,19 @@ subroutine ice_pio_initdecomp_4d (ndim3, ndim4, iodesc, precision)
    end subroutine ice_pio_initdecomp_4d
 
 
+!================================================================================
+
+   ! PIO Finalize
+
+   subroutine ice_pio_finalize()
+
+      integer(kind=int_kind)      :: ierr
+      character(len=*), parameter :: subname = '(ice_pio_finalize)'
+
+      call pio_finalize(ice_pio_subsystem,ierr)
```

Comment:
Mostly when there is an ierr code returned from a pio function, elsewhere there is a 

`call pio_seterrorhandling(File, PIO_RETURN_ERROR)`

before the call, then the ierr is checked with `ice_pio_check` and the error handling is set back to internal

Reply:
Good catch, I've updated the implementation.